Dockerfile

```diff
@@ -1,29 +1,53 @@
-FROM ubuntu:16.04
+FROM ubuntu:18.04
 WORKDIR /ardupilot
 
-RUN useradd -U -d /ardupilot ardupilot && \
+# Say that we won't anwser question
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN useradd -U -m ardupilot && \
     usermod -G users ardupilot
 
 RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install --no-install-recommends -y \
     lsb-release \
     sudo \
     software-properties-common \
-    python-software-properties
+    grep
+
+COPY Tools/environment_install/install-prereqs-ubuntu.sh /ardupilot/Tools/environment_install/
+
+RUN BASE_PKGS=$(grep --color=never -Po "^BASE_PKGS=\K.*" "Tools/environment_install/install-prereqs-ubuntu.sh" -m 1 | tr -d \") \
```

Comment:
This looks fragile.

You could do the same grepping within the current install script.